Assets/Scripts/Tools/Behaviour Tree/Nodes/EqualsNode.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using RPG;
+
+namespace BehaviourTree
+{
+    public class EqualsNode : IBehaviourTreeNode
+    {
+        private const string PROP_SOURCE = "source-property";
+        private const string PROP_NUMBER = "comparison-number";
+
+        public void Init(Behaviour behaviour)
+        {
+            behaviour.Properties.Add(PROP_SOURCE, new VariableProperty(VariableProperty.Type.String));
+            behaviour.Properties.Add(PROP_NUMBER, new VariableProperty(VariableProperty.Type.Number));
+        }
+
+        public NodeStatus Tick(Tree<Behaviour>.Node self, BehaviourObject obj)
+        {
+            Behaviour behaviour = self.Element;
+
+            string source = behaviour.GetProperty(PROP_SOURCE).GetString();
+            int num1 = (int)obj.GetProperty(source);
+            int num2 = (int)behaviour.GetProperty(PROP_NUMBER).GetNumber();
```

Comment:
I think I'll open an issue to make VariableProperties equatable so you don't have to do any gross casting. Thoughts?

Reply:
That would be awesome; in fact, making them Comparable would be even better since Greater Than and Less Than are almost certainly gonna pop up in the future 👌